src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -638,7 +640,7 @@ private bool TryInsert(TKey key, TValue value, InsertionBehavior behavior)
 
             if (updateFreeList)
             {
-                _freeList = entry.next;
+                _freeList = -3 - entries[_freeList].next;
```

Comment:
This is a bit weird; add comment on what it means?

Reply:
explained on top on next declaration

Reply:
Add a `const`?
```csharp
const int StartOfFreeList = -3;

// ...

_freeList = StartOfFreeList - entries[_freeList].next;
```

Reply:
thank's